src/Synth/ADnote.cpp

```diff
@@ -99,6 +103,63 @@ ADnote::ADnote(ADnoteParameters *pars_, const SynthParams &spars,
     memory.endTransaction();
 }
 
+int ADnote::fillOscilSmpFromWt(int nvoice)
+{
+    //The extra points contain the first points
+    std::size_t buf_alloc_size = synth.oscilsize + OSCIL_SMP_EXTRA_SAMPLES;
+
+    //Get the voice's oscil or external's voice oscil
+    int vc = nvoice;
+    if(pars.VoicePar[nvoice].Pextoscil != -1 && !waveTables)
+        vc = pars.VoicePar[nvoice].Pextoscil;
+    int oscposhi_start;
+
+    auto &voice = NoteVoicePar[nvoice];
+    assert(pars.VoicePar[vc].table);
+    WaveTable* wt = pars.VoicePar[vc].table;
+    Voice::OscilSmpT& OscilSmpToFill = voice.OscilSmp;
+    if(waveTables)
+    {
+        if(wt->mode() == wavetable_types::WtMode::freqwave_smps) {
+            OscilSmpToFill.table = wt;
+            OscilSmpToFill.isWaveTable = true;
+            // get the "center" around which we want to modulate
+            {
+                char bfp = pars.VoicePar[vc].OscilGn->Pbasefuncpar;
+                NoteVoicePar[nvoice].basefuncpar = (bfp == 64)
+                                                 ? 0.5f // fix to get exact middle
+                                                 : (((float)bfp) / 127.0f);
+            }
+        } else {
+            const float* bufferInTable = wt->getNextWaveToConsume(getvoicebasefreq(nvoice)).data();
+            voice.OscilSmp.smps = memory.valloc<float>(buf_alloc_size);
+            std::copy(bufferInTable, bufferInTable + synth.oscilsize, voice.OscilSmp.smps);
+            //Store the first elements to the last position for speedups
+            for(int i = 0; i < OSCIL_SMP_EXTRA_SAMPLES; ++i)
+                voice.OscilSmp.smps[synth.oscilsize + i] = voice.OscilSmp.smps[i];
+            voice.OscilSmp.isWaveTable = false;
+        }
+        oscposhi_start = pars.VoicePar[vc].OscilGn->getFinalOutpos();
```

Comment:
oscposhi_start (type int) is getting populated by getFinalOutpos() type char. For wavetables in length above 127 samples this seems possibly wrong?